src/shell/shell_cmds.cpp

```diff
@@ -496,6 +497,49 @@ static std::vector<int> map_name_lengths(const std::vector<DtaResult> &results,
 	return ws;
 }
 
+static std::vector<int> calc_column_widths(const std::vector<int> &cell_widths,
+                                           int min_cell_width)
+{
+	assert(min_cell_width > 0);
+
+	// Actual text columns using current text mode;
+	// in practice it's either 40, 80, or 132.
+	const int tcols = real_readw(BIOSMEM_SEG, BIOSMEM_NB_COLS);
+
+	// Use tcols-1 because we never want to print line up to the actual
+	// limit; this would cause unnecessary line wrapping
+	const size_t max_columns = (tcols - 1) / min_cell_width;
+	std::vector<int> col_widths(max_columns);
+
+	// This function returns true when column number is too high to fit
+	// words into a terminal width.  If it returns false, then the first
+	// coln integers in col_widths vector describe the column widths.
+	auto too_many_columns = [&](size_t coln) -> bool {
+		std::fill(col_widths.begin(), col_widths.end(), 0);
+		if (coln <= 1)
+			return false;
+		int max_line_width = 0; // tally of the longest line
+		int c = 0;              // current columnt
+		for (const int width : cell_widths) {
+			const int old_col_width = col_widths[c];
+			const int new_col_width = std::max(old_col_width, width);
+			col_widths[c] = new_col_width;
+			max_line_width += (new_col_width - old_col_width);
+			if (max_line_width >= tcols)
+				return true;
+			c = (c + 1) % coln;
```

Comment:
`col_widths` (and old and new), `too_many_columns`, `max_line_width` --> these are great names, and I can follow.

Things break down again w/ `tcols` and `cell_widths`; so I think this lamda will be fixed up once these two are adjusted in the prior comment.